src/com/spikes2212/dashboard/ConstantHandler.java

```diff
@@ -28,11 +35,16 @@
      * This will be either the value got in the function call, or the value given by the user on the {@link SmartDashboard}.
      * If it cannot communicate a value from the {@link SmartDashboard}, it will return a 0.
      */
+	
     public static Supplier<Double> addConstantDouble(String name, double value) {
-        SmartDashboard.putNumber(name, value);
-        return () -> SmartDashboard.getNumber(name, value);
+        if (!Preferences.getInstance().containsKey(name)) {
+        	Preferences.getInstance().putDouble(name, value);
+        	doubleMap.put(name, value);

```

Comment:
you should always remember your hard coded values, not only if they don't exist in preferences